### ISSUE TITLE
Textmate grammar support for indexed properties 

### ISSUE THREAD

**Issue by xorye** [redhat-developer/vscode-quarkus#283]
For property keys that contain symbols like [ and #, the properties textmate grammar displays incorrect highlighting. 

For example, for `openshift.config-map-volumes[0].config-map-name`, the property key is not highlighted in blue:

![image](https://user-images.githubusercontent.com/20326645/87319332-7e1cbc00-c4f7-11ea-94cc-7004c7ca2496.png)

For properties that contain #:
![image](https://user-images.githubusercontent.com/20326645/87593192-11e9b600-c6b9-11ea-95a9-71558704be8f.png)

Currently, the TextMate grammar assumes that property keys are only composed of: `\/a-zA-Z0-9%_.-`https://github.com/redhat-developer/vscode-quarkus/blob/7036c9dbe90e2dae8596d84318e6faff0f85bdad/language-support/properties-support/java-properties.tmLanguage.json#L41

**Comment by datho7561** [comment]
WIP:

![Screenshot from 2020-07-28 16-56-26](https://user-images.githubusercontent.com/22376627/88720793-72361a00-d0f3-11ea-8b4a-1525a3a6d517.png)

Should I keep the highlighting for the index, or change it back to the same colour as the rest of the property name (since this will be in vscode-microprofile).

**Comment by datho7561** [comment]
Here is what I settled on. I used the default VSCode theme (Dark+) in this screenshot to avoid confusion regarding the colours:

![Screenshot from 2020-07-29 12-04-45](https://user-images.githubusercontent.com/22376627/88823976-b678f700-d193-11ea-996b-958a672d035c.png)

Properties with square braces in their names have no special meaning in most implementations of MicroProfile Config as far I can tell, so it doesn't make sense to do specific highlighting. If this feature is helpful for Quarkus, we could maybe do it in vscode-quarkus.
